lib/checkVerificationCode.ts

```diff
@@ -0,0 +1,32 @@
+import { FetchResult } from "apollo-link";
+import gql from "graphql-tag";
+
+import { CheckVerificationCodeResult } from "@lib/@types/VerificationCode";
+import { fetchManagement } from "@src/utils/fetchManagement";
+
+const CHECK_VERIFICATION_CODE_QUERY = gql`
+  query checkVerificationCodeQuery($code: String!, $eventId: String!) {
+    checkVerificationCode(input: { code: $code, eventId: $eventId }) {
+      identityType
+      identityValue
+      nonce
+      status
+    }
+  }
+`;
+
+export async function checkVerificationCode(
+  code: string,
+  eventId: string,
+): Promise<
+  FetchResult<{ checkVerificationCode: CheckVerificationCodeResult }>
+> {
+  const operation = {
+    query: CHECK_VERIFICATION_CODE_QUERY,
+    variables: { code, eventId },
+  };
+
+  return (await fetchManagement(operation)) as FetchResult<{
+    checkVerificationCode: CheckVerificationCodeResult;
+  }>;
```

Comment:
This clear looks like a `return await` but I guess you can't do the `as FetchResult<...>` without it.
What do you think of storing the result in a variable that you type and then simply return this variable?

Reply:
better yet, what about making?
```typescript
fetchManagement<{
     checkVerificationCode: CheckVerificationCodeResult;
   }>(operation)
```

Reply:
I wanted to make `fetchManagenement()` typed with a generic a few weeks ago, but it looks like a complex task:

```ts
Type 'Promise<FetchResult<{ [key: string]: any; }, Record<string, any>, Record<string, any>>>' is not assignable to type 'Promise<FetchResult<T, Record<string, any>, Record<string, any>>>'.
  Type 'FetchResult<{ [key: string]: any; }, Record<string, any>, Record<string, any>>' is not assignable to type 'FetchResult<T, Record<string, any>, Record<string, any>>'.
    Type 'FetchResult<{ [key: string]: any; }, Record<string, any>, Record<string, any>>' is not assignable to type 'ExecutionResult<T>'.
      Types of property 'data' are incompatible.
        Type '{ [key: string]: any; } | null | undefined' is not assignable to type 'T | null | undefined'.
          Type '{ [key: string]: any; }' is not assignable to type 'T'.
            'T' could be instantiated with an arbitrary type which could be unrelated to '{ [key: string]: any; }'.ts(2322)
```

Reply:
oook 😅